File: Python3/Easy/RuntimeError/longest-common-prefix_2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        if not strs:
            return ""
        if len(strs) == 1:
            return strs[0]
        prefix = strs[0]
        for i in range(1, len(strs)):
            prefix = self.find_prefix(prefix, strs[i])
        return prefix

File: Rust/Easy/Accepted/longest-common-prefix_2.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        
        let mut prefix = String::new();
        let mut i = 0;
        loop {
            let mut c = None;
            for s in &strs {
                if i >= s.len() {
                    return prefix;
                }
                let cc = s.chars().nth(i).unwrap();
                if c.is_none() {
                    c = Some(cc);
                } else if c.unwrap() != cc {
                    return prefix;
                }
            }
            prefix.push(c.unwrap());
            i += 1;
        }
    }
}